core/src/test/java/org/elasticsearch/action/update/UpdateResponseTests.java

```diff
@@ -0,0 +1,137 @@
+        }
+        {
+            BytesReference source = new BytesArray("{\"title\":\"Book title\",\"isbn\":\"ABC-123\"}");
+            Map<String, GetField> fields = new HashMap<>();
+            fields.put("title", new GetField("title", Collections.singletonList("Book title")));
+            fields.put("isbn", new GetField("isbn", Collections.singletonList("ABC-123")));
+
+            UpdateResponse updateResponse = new UpdateResponse(new ReplicationResponse.ShardInfo(3, 2),
+                    new ShardId("books", "books_uuid", 2), "book", "1", 7, 2, UPDATED);
+            updateResponse.setGetResult(new GetResult("books", "book", "1", 2, true, source, fields));
+
+            String output = Strings.toString(updateResponse);
+            assertEquals("{\"_index\":\"books\",\"_type\":\"book\",\"_id\":\"1\",\"_version\":2,\"result\":\"updated\"," +
+                    "\"_shards\":{\"total\":3,\"successful\":2,\"failed\":0},\"_seq_no\":7,\"get\":{\"found\":true," +
+                    "\"_source\":{\"title\":\"Book title\",\"isbn\":\"ABC-123\"},\"fields\":{\"isbn\":[\"ABC-123\"],\"title\":[\"Book " +
+                    "title\"]}}}", output);
+        }
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+        final Tuple<UpdateResponse, UpdateResponse> tuple = randomUpdateResponse();
+
+        // Parse the XContent bytes to obtain a parsed UpdateResponse
+        UpdateResponse parsedUpdateResponse;
+        try (XContentParser parser = createParser(xContentType.xContent(), toXContent(tuple.v1(), xContentType))) {
+            parsedUpdateResponse = UpdateResponse.fromXContent(parser);
+            assertNull(parser.nextToken());
+        }
+
+        final UpdateResponse updateResponse = tuple.v2();
+        try (XContentParser parser = createParser(xContentType.xContent(), toXContent(parsedUpdateResponse, xContentType))) {
+            IndexResponseTests.assertDocWriteResponse(updateResponse, parser.map());
+        }
+        assertEquals(updateResponse.getGetResult(), parsedUpdateResponse.getGetResult());
+    }
+
+    private static Tuple<UpdateResponse, UpdateResponse> randomUpdateResponse() {
+        Tuple<GetResult, GetResult> getResults = GetResultTests.randomGetResult(randomFrom(XContentType.values()));
+        GetResult actualGetResult = getResults.v1();
+
+        ShardId shardId = new ShardId(actualGetResult.getIndex(), randomAsciiOfLength(5), randomIntBetween(0, 5));
+        String type = actualGetResult.getType();
+        String id = actualGetResult.getId();
+        long version = actualGetResult.getVersion();
+        DocWriteResponse.Result result = actualGetResult.isExists() ? DocWriteResponse.Result.UPDATED : DocWriteResponse.Result.NOT_FOUND;
+
+        Long seqNo = randomBoolean() ? null : randomNonNegativeLong();
```

Comment:
Generating random longs like this tends to generate really large numbers. I added another 50/50 condition in some of my randomized tests that do e.g. a %10000 to get smaller numbers here more often, this uncovered some type conversion issues (integer/long) in the parser. Maybe you could add something like that here too, since smaller numbers are more likely to appear in real responses.